tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -862,3 +820,59 @@ def brick_status_alert(hostname):
     finally:
         if isinstance(lock, etcd.lock.Lock) and lock.is_acquired:
             lock.release()
+
+
+def update_cluster_alert_count():
+    cluster_alert_count = 0
+    severity = ["WARNING", "CRITICAL"]
+    try:
+        alert_counts = find_volume_id()
+        alerts_arr = NS.tendrl.objects.ClusterAlert(
+            tags={'integration_id': NS.tendrl_context.integration_id}
+        ).load_all()
+        for alert in alerts_arr:
+            alert.tags = json.loads(alert.tags)
+            if alert.severity in severity:
+                cluster_alert_count += 1
+                if alert.resource in NS.gluster.objects.VolumeAlertCounters(
+                        )._defs['relationship'][alert.alert_type.lower()]:
+                    vol_name = alert.tags.get('volume_name', None)
+                    if vol_name:
+                        if vol_name in alert_counts.keys():
+                            alert_counts[vol_name]['alert_count'] += 1
+        # Update cluster alert count
+        NS.tendrl.objects.ClusterAlertCounters(
+            integration_id=NS.tendrl_context.integration_id,
+            alert_count=cluster_alert_count
+        ).save()
+        # Update volume alert count
+        for volume in alert_counts:
+            NS.gluster.objects.VolumeAlertCounters(
+                integration_id=NS.tendrl_context.integration_id,
+                alert_count=alert_counts[volume]['alert_count'],
+                volume_id=alert_counts[volume]['vol_id']
+            ).save()
+    except etcd.EtcdException as ex:
+        logger.log(
+            "debug",
+            NS.publisher_id,
+            {"message": "Unable to update alert count.err: %s" % ex}
+        )
+
+
+def find_volume_id():
+    alert_counts = {}
+    volumes = etcd_utils.read(
```

Comment:
Why not use `NS.gluster.objects.Volume().load_all()` here ? I feel all you need is volume names then form the dict.
Also call the function as `get_volume_alert_counts`

Reply:
ok

Reply:
i am just reading, volume_id and forming a dict with alert_count is 0, how can we call this function get_volume_alert_counts?